daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala

```diff
@@ -252,141 +314,200 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
           e"""\(sig : Party) (obs : Party) ->
                 let c: M:T = M:T { signatory = sig, observer = obs, precondition = True, key = M:toKey sig, nested = M:buildNested 0 }
                 in ubind x : ContractId M:T <- create @M:T c
-                  in create @M:T c
+                  in Test:create c
            """,
           Array(SParty(alice), SParty(bob)),
-          alice,
+          Set(alice),
         )
         inside(res) { case Success(Left(SErrorDamlException(IE.DuplicateContractKey(_)))) =>
-          msgs shouldBe createTraces :++ createTraces
+          msgs shouldBe createTraces.dropRight(1)
         }
       }
 
       // TEST_EVIDENCE: Semantics: Evaluation order of create with empty contract key maintainers
       "empty contract key maintainers" in {
         val (res, msgs) = evalUpdateApp(
           pkgs,
-          e"""\(sig : Party) (obs : Party) -> create @M:T
-              M:T { signatory = sig, observer = obs, precondition = True, key = M:keyNoMaintainers, nested = M:buildNested 0 }
+          e"""\(sig : Party) (obs : Party) -> 
+                Test:create M:T { signatory = sig, observer = obs, precondition = True, key = M:keyNoMaintainers, nested = M:buildNested 0 }
            """,
           Array(SParty(alice), SParty(bob)),
-          alice,
+          Set(alice),
         )
         inside(res) {
           case Success(
                 Left(SErrorDamlException(IE.CreateEmptyContractKeyMaintainers(T, _, _)))
               ) =>
-            msgs shouldBe createTraces
+            msgs shouldBe createTraces.dropRight(1)
         }
       }
 
       // TEST_EVIDENCE: Semantics: Evaluation order of create with authorization failure
       "authorization failure" in {
         val (res, msgs) = evalUpdateApp(
           pkgs,
-          e"""\(sig : Party) (obs : Party) -> create @M:T
-              M:T { signatory = sig, observer = obs, precondition = True, key = M:toKey sig, nested = M:buildNested 0 }
+          e"""\(sig : Party) (obs : Party) -> 
+                Test:create M:T { signatory = sig, observer = obs, precondition = True, key = M:toKey sig, nested = M:buildNested 0 }
            """,
           Array(SParty(alice), SParty(bob)),
-          bob,
+          Set(bob),
         )
         inside(res) { case Success(Left(SErrorDamlException(IE.FailedAuthorization(_, _)))) =>
-          msgs shouldBe createTraces
+          msgs shouldBe createTraces.dropRight(1)
```

Comment:
maybe define that as a synonym somewhere. It’s not immediately obvious from reading it what it does.